desc/equilibrium/equilibrium.py

```diff
@@ -858,6 +858,46 @@ def compute(  # noqa: C901
             msg="must pass in a Grid object for argument grid!"
             f" instead got type {type(grid)}",
         )
+        # a check for Redl to prevent computing on-axis or
+        # at a rho=1.0 point where profiles vanish
+        if (
+            any(["Redl" in name for name in names])
+            and self.electron_density is not None
+        ):
+            warnif(
+                grid.axis.size,
+                UserWarning,
+                "Redl formula is undefined at rho=0, "
+                "but grid has grid points at rho=0, note that on-axis"
+                "current will be NaN.",
+            )
+            rho = grid.compress(grid.nodes[:, 0], "rho")
+
+            # check if profiles may go to zero
+            # if they are exactly zero this would cause NaNs since the profiles
+            # vanish.
+            warnif(
+                np.any(np.isclose(self.electron_density(rho), 0.0, atol=1e-8)),
```

Comment:
do we need atol here? I thought the issue was only if the profile is actually zero. If its small but finite I thought it should work? (though accuracy may be degraded)

Reply:
True, I put atol here considering if at some point the density is lower than 1e-8, it is more likely that we are close to an actual zero of the profile, rather than just near a point of small but non-vanishing electron density.

This of course may be a poor assumption but until it becomes an issue later I am fine with keeping this as is.